Integrated all Module 7 subchapters into Quarto website (#204)
UBC-MDS/exploratory-data-viz: modules/module7/slides/module7_07.qmd

</text>

:::{.notes}
Given that displaying the JSON spec correctly depends on a the Vega-Lite web libraries,
our charts will only render if we have an internet connection.

If you are working offline and want to render charts in JupyterLab,
there are a few different options.

If you want interactive chart elements to work,
you need to install a JupyterLab extension
that downloads the the Vega-Lite rendering library to your machine.
[The steps for installing this extension are outlined in the Altair documentation](https://altair-viz.github.io/user_guide/display_frontends.html#displaying-in-jupyterlab)

If you are happy with seeing a static version of your charts when you are offline,
you could instruct Altair to use the Vega-Lite libraries
only if there is an active internet connection
and otherwise use and image for the plot.

To do this,
you can include the code from this slide on top of your notebook,
just after your import statements.
This will not cause any changes to the appearance of your chart.
:::

## What if we want to save our charts as images instead?

> - The three dots action button can be used to save images manually.
> - `altair_saver` enables saving of PNG and SVG charts .
>     - [Instructions for how to install and work with Altair saver can be found here](https://github.com/altair-viz/altair_saver/).
>     - The syntax is the same as before `chart.save('my-chart.png')`
> 
> :::{.notes}
> If we want to save a chart as a PNG image file
> we can use the three dot action button
> as we saw in the very first module of this course.
> When we are clicking the "Save as PNG" entry in this menu,
> the browser converts the Vega-Lite spec into a PNG image
> that is downloaded to our computer.
> 
> If we want to save an image as a PNG programatically
> we need to use a helper package called `altair_saver`.
> This package let's Altair access the save functionality of the browser
> in a similar way as when you click the thee dot action button.
> 
> [Instructions for how to install
> and work with Altair saver can be found here](https://github.com/altair-viz/altair_saver/).
> After it is installed,
> you can save as PNG by typing `chart.save('my-chart.png')`.
> :::

Reviewer: ```suggestion
- The three dots action button can be used to save charts as PNG or SVG manually.
- Charts can also be saved programmatically via the same syntax as before, e.g. `chart.save('my-chart.pdf')`

:::{.notes}
If we want to save a chart as a PNG or SVG file
we can use the three dot action button
as we saw in the very first module of this course.
When we are clicking the "Save as PNG" entry in this menu,
the browser converts the Vega-Lite spec into a PNG image
that is downloaded to our computer.

If we want to save charts programmatically as either PNG, SVG, or PDF
we need to use a helper package called `vl-convert`.
This package let's Altair access the save functionality of the browser
in a similar way as when you click the thee dot action button.

[More details on how to save charts and install vl-convert can be found in the Altair documentation](https://altair-viz.github.io/user_guide/saving_charts.html#png-svg-and-pdf-format).
:::
```

> ## What is the difference between saving in different formats?
> 
> - PNG is a bitmap/raster format
>     - Use to store data in a fixed grid of pixels
>     - Good for photos and complex illustrations
>     - Zooming in causes the image to become blurry/pixelated
>     - The file size depends on the number of pixels
>     
> - SVG is a scalable graphics / vector format
>     - Stores data as text with mathematical formulas
>     - Good for simpler illustration
>     - Zooming in retains perfect quality
>     - The file size depends on the number of objects and their complexity
> 
> :::{.notes}
> In additional to saving as PNG,
> we could also have saved as SVG
> by changing the file extension accordingly.
> 
> What is the difference between these two options?<br><br>
> **PNG is a bitmap/raster format**<br>
>     - Stores data in a grid of pixels of different colors<br>
>     - Zooming in causes the image to become blurry/pixelated<br>
>     - The file size depends on the number of pixels<br>
>     - Often ideal for photos and complex illustrations<br><br>
> **SVG is a scalable graphics / vector format**<br>
>     - Stores data as text by describing objects with mathematical formulas<br>
>     - Zooming in retains perfect quality<br>
>     - The file size depends on the number of objects and their complexity<br>
>     - Often ideal for simpler illustration such as most charts<br><br>
> Saving as HTML has similar advantages and disadvantages as SVG.<br>
> In fact HTML uses SVG for creating the graphics.
> 
> Another common format is PDF,
> which can be a combination of raster images and scalable graphics formats.
> `altair_saver` allows you to save as pdf,
> but you have to follow specific installation steps in the instructions
> we linked to in the last slide.

Reviewer: ```suggestion
## What is the difference between saving in different formats?

- PNG is a bitmap/raster format
    - Use to store data in a fixed grid of pixels
    - Good for photos and complex illustrations
    - Zooming in causes the image to become blurry/pixelated
    - The file size depends on the number of pixels
    
- SVG and PDF are a scalable graphics / vector formats
    - Stores data as text with mathematical formulas
    - Good for simpler illustration
    - Zooming in retains perfect quality
    - The file size depends on the number of objects and their complexity

:::{.notes}
In additional to saving as PNG,
we could also have saved as SVG or PDF
by changing the file extension accordingly.

What is the difference between these two options?<br><br>
**PNG is a bitmap/raster format**<br>
    - Stores data in a grid of pixels of different colors<br>
    - Zooming in causes the image to become blurry/pixelated<br>
    - The file size depends on the number of pixels<br>
    - Often ideal for photos and complex illustrations<br><br>
**SVG and PDF are scalable graphics / vector formats**<br>
    - Stores data as text by describing objects with mathematical formulas<br>
    - Zooming in retains perfect quality<br>
    - The file size depends on the number of objects and their complexity<br>
    - Often ideal for simpler illustration such as most charts<br><br>
Saving as HTML has similar advantages and disadvantages as SVG.<br>
In fact HTML uses SVG for creating the graphics.
```
